ttSF/write_cards_ttCR.py

```diff
@@ -0,0 +1,391 @@
+    global opt, args
+    (opt, args) = parser.parse_args()
+
+
+def  writeCard(backgrounds,signals,select,region=-1) :
+
+    if   "0b0j"         in select : theCat = "0"
+    elif "1b1j"         in select : theCat = "1"
+    elif "2b0j"         in select : theCat = "2"
+    elif "boosted"      in select : theCat = "3"
+    elif "ttCR_invMcut" in select : theCat = "4"
+
+    variable = {
+        "0"  : "DNNoutSM_kl_1",
+        "1"  : "DNNoutSM_kl_1",
+        "2"  : "DNNoutSM_kl_1",
+        "3"  : "DNNoutSM_kl_1",
+        "4"  : "DNNoutSM_kl_1",
+    }
+
+    theOutputDir = "{0}{1}".format(select,variable[theCat])
+    dname = "_"+opt.channel+opt.outDir
+    out_dir = "cards{1}/{0}/".format(theOutputDir,dname)
+
+    cmd = "mkdir -p {0}".format(out_dir)
+        
+    regionName = ["","regB","regC","regD"]
+    if theCat == '4': regionName = [""]
+    regionSuffix = ["SR","SStight","OSinviso","SSinviso"]
+    if theCat == '4': regionSuffix = ["SR"]
+    status, output = commands.getstatusoutput(cmd)
+    thechannel = "2"
+    if opt.channel == "ETau" : thechannel="1"
+    elif opt.channel == "MuTau" : thechannel = "0"
+    elif opt.channel == "inclusive": thechannel = "3"
+
+    #read config
+    categories = []
+    categories.append((0,select))
+    MCbackgrounds=[]
+    processes=[]
+    inRoot = TFile.Open(opt.filename)
+    for bkg in backgrounds:
+        #Add protection against empty processes => If I remove this I could build all bins at once instead of looping on the selections
+        templateName = "{0}_{1}_SR_{2}".format(bkg,select,variable[theCat])
+        template = inRoot.Get(templateName)
+        if template.Integral()>0.000001 :
+            processes.append(bkg)
+            if bkg is not "QCD" :
+                MCbackgrounds.append(bkg)
+
+    rates = []
+    iQCD = -1
+    totRate = 0
+    templateName = "data_obs_{0}_{2}_{1}".format(select,variable[theCat],regionSuffix[region])
+    template = inRoot.Get(templateName)
+    obs = template.GetEntries()
+
+    for proc in range(len(backgrounds)):
+        if "QCD" in backgrounds[proc] and region > 0:
+            rates.append(-1)
+            iQCD = proc
+        else :
+            templateName = "{0}_{1}_{3}_{2}".format(backgrounds[proc],select,variable[theCat],regionSuffix[region])
+            template = inRoot.Get(templateName)
+            brate = template.Integral()
+            rates.append(brate)
+            totRate = totRate + brate
+    if iQCD >= 0 and region > 0: rates[iQCD] = TMath.Max(0.0000001,obs-totRate)
+    print str(region), rates
+
+    if region == 0 :
+        for proc in range(len(signals)):
+            templateName = "{0}_{1}_{3}_{2}".format(signals[proc],select,variable[theCat],regionSuffix[region])
+            template = inRoot.Get(templateName)
+            srate = template.Integral()
+            rates.append(srate)
+
+        syst = systReader("../config/MY_systematics_ttCR.cfg",signals,backgrounds,None)
```

Comment:
tìyou sould add this config which is missing

Reply:
I have added it, but it is just an empty file because we decided not to insert any syst in the CR. If we decide to add some systs to the CR then there it's where they go

Reply:
(I have also taken away the MY_ which is the placeholder I use to identify my personal mods to the code)

Reply:
Then please add at least one commented line explaining that for the moment we do not apply any systematic.

Reply:
ok

Reply:
Sorry, I meant to add the commented lines inside the `systematics_ttCR.cfg` file so it's not completely empty.